### ISSUE TITLE
Create audit log for our trial deduplication process

### ISSUE THREAD

**Issue by vitorbaptista** [opentrials/opentrials#676]
Now that we're improving our trial deduplication methods, we must be able to know why a trial was deduplicated, so we can debug our processes.

The way I envision it is adding a table to the API database (so we can add it to the API results in the future) similar to:

| trial_id | record_id (unique) | method | commit | created_at | updated_at |
| --- | --- | --- | --- | --- | --- |
| A | X | identifier | COMMIT_HASH | | |
| A | Y | title | COMMIT_HASH | | |

I haven't deeply analysed this though, so feel free to propose another way. This would also require us to somehow add the `COMMIT_HASH` to our processors, I'm not sure how.

# Acceptance Criteria

* I can see why, when and how each record was deduplicated
* I can see what code version did the deduplication

**Comment by arthurSena** [comment]
@vitorbaptista We've been thinking about this solution and we'd like to know if we must implement some priority on `method` column.
 For example: The log of a trial that was just matched by title should not be updated by the log of a match occured via indentifiers as we supose that title matching is more relevant. Also, we think that eventually all values on `method` column will be `via identifiers` as the processors are executed multiple times. Antother approach is creating multiple lines in order to store more than one method related to the same `record_id` as you can see in the table below.

trial_id | record_id | method | commit | created_at | updated_at
------------ | -------------|------------ | -------------|------------ | -------------
A | B|identifier | 
A | B|title |

**Comment by vitorbaptista** [comment]
@arthurSena The table should be updated every time we change the match. In this sense, it would answer "How each record was matched on its ***current*** trial?", as in we won't know if a record was matched to another trial previously.

It would be good to have the entire log, including previous matches, but I'd like to avoid writing to the audit log table every time we process the trials, even if it kept the same match. Like:

trial_id | record_id | method | commit | created_at | updated_at
------------ | -------------|------------ | -------------|------------ | -------------
A | B|title | | 2017-02-01 | |
A | B|identifier |  | 2017-02-05 | |

And not like:

trial_id | record_id | method | commit | created_at | updated_at
------------ | -------------|------------ | -------------|------------ | -------------
A | B|title | | 2017-02-01 | |
A | B|title | | 2017-02-02 | |
A | B|title | | 2017-02-03 | |
A | B|title | | 2017-02-04 | |
A | B|identifier |  | 2017-02-05 | |
A | B|identifier |  | 2017-02-06 | |
A | B|identifier |  | 2017-02-07 | |
... | | |  |  | |

In other words, log every time a different match is made. So if the match was still for `(A, B, title)` (the same `(trial_id, record_id, method)`), don't write a new row. Otherwise write a new row.

What do you think?

**Comment by arthurSena** [comment]
@vitorbaptista We think that this solution is sensible and already have it implemented.
However we'd need the [other PR](https://github.com/opentrials/processors/pull/115) to be accepted before we send this one,
as we have been working on this issue in a branch that derives from the PR one.
